manager/apis/app/v1beta1/fybrikapplication_types.go

```diff
@@ -0,0 +1,240 @@
+	// +optional
+	ErrorMessage string `json:"errorMessage,omitempty"`
+
+	// AssetStates provides a status per asset
+	// +optional
+	AssetStates map[string]AssetState `json:"assetStates,omitempty"`
+
+	// ObservedGeneration is taken from the FybrikApplication metadata.  This is used to determine during reconcile
+	// whether reconcile was called because the desired state changed, or whether the Blueprint status changed.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// ValidatedGeneration is the version of the FyrbikApplication that has been validated with the taxonomy defined.
+	// +optional
+	ValidatedGeneration int64 `json:"validatedGeneration,omitempty"`
+
+	// ValidApplication indicates whether the FybrikApplication is valid given the defined taxonomy
+	// +optional
+	ValidApplication corev1.ConditionStatus `json:"validApplication,omitempty"`
+
+	// Generated resource identifier
+	// +optional
+	Generated *ResourceReference `json:"generated,omitempty"`
+
+	// ProvisionedStorage maps a dataset (identified by AssetID) to the new provisioned bucket.
+	// It allows FybrikApplication controller to manage buckets in case the spec has been modified, an error has occurred,
+	// or a delete event has been received.
+	// ProvisionedStorage has the information required to register the dataset once the owned plotter resource is ready
+	// +optional
+	ProvisionedStorage map[string]DatasetDetails `json:"provisionedStorage,omitempty"`
+}
+
+// FybrikApplication provides information about the application whose data is being operated on,
+// the nature of the processing, and the data sets chosen for processing by the application.
+// The FybrikApplication controller obtains instructions regarding any governance related changes that must
+// be performed on the data, identifies the modules capable of performing such changes, and finally
+// generates the Plotter which defines the secure runtime environment and all the components
+// in it.  This runtime environment provides the application with access to the data requested
+// in a secure manner and without having to provide any credentials for the data sets.  The credentials are obtained automatically
+// by the manager from the credential management system.
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+type FybrikApplication struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec   FybrikApplicationSpec   `json:"spec"`
+	Status FybrikApplicationStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// FybrikApplicationList contains a list of FybrikApplication
+type FybrikApplicationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []FybrikApplication `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&FybrikApplication{}, &FybrikApplicationList{})
+}
+
+const (
```

Comment:
These definitions should be removed from v1alpha1 package.

Reply:
The better to move all this constants to controllers, they are not part of APIs.

Reply:
They can't be in the controllers package because of the import loop, maybe somewhere under pkg/